Adds Core Web Vitals tools (#3048)
GoogleChrome/web.dev: src/site/content/en/blog/vitals-tools/index.md

hero: hero.png
thumbnail: thumbnail.png
alt: Chrome User Experience logo, PageSpeed Insights logo, Lighthouse logo, Search Console logo, DevTools logo, Web Vitals extension logo.
tags:
  - blog # blog is a required tag for the article to show up in the blog.
  - fast
---

The recently announced [Web Vitals](/vitals/) initiative provides unified guidance about quality signals that are essential for all sites to deliver a great user experience on the web. We're happy to announce that **all of our popular tools for web developers now support measurement of Core Web Vitals**, helping you more easily diagnose and fix user experience issues. This includes [Lighthouse](https://github.com/GoogleChrome/lighthouse), [PageSpeed Insights](https://developers.google.com/speed/pagespeed/insights/), [Chrome DevTools](https://developers.google.com/web/tools/chrome-devtools), [Search Console](https://search.google.com/search-console/about), the [Web Vitals Chrome extension](https://chrome.google.com/webstore/detail/web-vitals/ahfhijdlegdabablpippeagghigmibma) and a new (!) [Chrome UX Report](https://developers.google.com/web/tools/chrome-user-experience-report) API. 

> 

Author: HOLD for addition of new paragraph 👋

---
title: "Tools to measure Core Web Vitals"
subhead: Your favorite developer tools can now measure the Core Web Vitals metrics.
authors:
  - addyosmani
  - egsweeny
date: 2020-05-28
description: |
  Read about the newly announced Core Web Vitals measurement support across popular web developer tools like Lighthouse, PageSpeed Insights, Chrome UX Report, and many others.

> hero: hero.png

Reviewer: I accidentally optimized this but I checked it on staging and it seems fine. If it ends up looking grainy we can just re-upload a higher fidelity version.

P.S. what's up with the 3 straight lines on the DevTools logo 😆 

![image](https://user-images.githubusercontent.com/4713486/83095264-c66f4000-a057-11ea-940b-31ff26841165.png)

Author: I'll correct this in my next push. Something funky happened with the SVG import first time around :)

thumbnail: thumbnail.png
alt: Chrome User Experience logo, PageSpeed Insights logo, Lighthouse logo, Search Console logo, Chrome DevTools logo, Web Vitals extension logo.
tags:
  - blog # blog is a required tag for the article to show up in the blog.
  - web-vitals
  - performance
---

> The recently announced [Web Vitals](/vitals/) initiative provides unified guidance about quality signals that are essential for all sites to deliver a great user experience on the web. We're happy to announce that **all of Google's popular tools for web developers now support measurement of Core Web Vitals**, helping you more easily diagnose and fix user experience issues. This includes [Lighthouse](https://github.com/GoogleChrome/lighthouse), [PageSpeed Insights](https://developers.google.com/speed/pagespeed/insights/), [Chrome DevTools](https://developers.google.com/web/tools/chrome-devtools), [Search Console](https://search.google.com/search-console/about), the [Web Vitals Chrome extension](https://chrome.google.com/webstore/detail/web-vitals/ahfhijdlegdabablpippeagghigmibma) and a new (!) [Chrome UX Report](https://developers.google.com/web/tools/chrome-user-experience-report) API. 

Reviewer: FYI I changed `all of our popular tools` to `all of Google's popular tools`

Once you identify a type of page that has Core Web Vitals related issues, you can use PageSpeed Insights to learn about specific optimization suggestions for representative pages.

### Web Vitals extension

The Web Vitals extension measures the three Core Web Vitals metrics in real-time for (desktop) Google Chrome. This is helpful for catching issues early on during your development workflow and as a diagnostic tool to assess performance of Core Web Vitals as you browse the web. 

The extension is now available to install from the [Chrome Web Store](https://chrome.google.com/webstore/detail/web-vitals/ahfhijdlegdabablpippeagghigmibma?hl=en)! We hope you find it useful. We welcome any contributions to improve it as well as feedback on the project's [GitHub](https://github.com/GoogleChrome/web-vitals-extension/) repository.

<figure class="w-figure">
  <img class="w-screenshot" src="./Vitals-Tools10.png" alt="Core Web Vitals displayed in real-time with the Web Vitals Chrome Extension">
</figure>

> #### Quick Highlights

Reviewer: ```suggestion
#### Quick highlights
```

  - web-vitals
  - performance
---

The recently announced [Web Vitals](/vitals/) initiative provides unified guidance about quality signals that are essential for all sites to deliver a great user experience on the web. We're happy to announce that **all of Google's popular tools for web developers now support measurement of Core Web Vitals**, helping you more easily diagnose and fix user experience issues. This includes [Lighthouse](https://github.com/GoogleChrome/lighthouse), [PageSpeed Insights](https://developers.google.com/speed/pagespeed/insights/), [Chrome DevTools](https://developers.google.com/web/tools/chrome-devtools), [Search Console](https://search.google.com/search-console/about), [web.dev's measure tool](/measure/), the [Web Vitals Chrome extension](https://chrome.google.com/webstore/detail/web-vitals/ahfhijdlegdabablpippeagghigmibma) and a new (!) [Chrome UX Report](https://developers.google.com/web/tools/chrome-user-experience-report) API. 

<figure class="w-figure">
  <img class="w-screenshot" src="./Vitals-Tools1.png" alt="Summary of Chrome and Search Tools that support the Core Web Vitals metrics">
</figure>

> **_[Quick Refresher]_** Lab tools provide insight into how a _potential user_ will likely experience your website and offer reproducible results for debugging. **Field** tools provide insight into how your _real users_ are experiencing your website; this type of measurement is often called Real User Monitoring (RUM). Each [lab or field tool](/how-to-measure-speed/#lab-data-vs-field-data) offers distinct value for optimizing your user experience. 
> 
> To begin your journey optimizing user-experience with Core Web Vitals, try the following workflow: 

Reviewer 2: ```suggestion
{% Aside 'key-term' %}
Lab tools provide insight into how a _potential user_ will likely experience your website and offer reproducible results for debugging. 
{% endAside %}

{% Aside 'key-term' %}
Field tools provide insight into how your _real users_ are experiencing your website; this type of measurement is often called Real User Monitoring (RUM). 
{% endAside %}

Each [lab or field tool](/how-to-measure-speed/#lab-data-vs-field-data) offers distinct value for optimizing your user experience. To begin your journey optimizing user-experience with Core Web Vitals, try the following workflow: 
```

Reviewer 2: "Quick refresher" seems arbitrary, I think key term aside formatting would serve well here :) https://web.dev/handbook/web-dev-components/#key-term-asides

Author: Thanks! I'll talk to Elizabeth about us switching over to the key term formatting before we land this :)

* Need a custom dashboard of Core Web Vitals? Use the updated CrUX Dashboard or new Chrome UX Report API for field data or PageSpeed Insights API for lab data.
* Looking for guidance? web.dev/measure can measure your page and show you a prioritized set of guides and codelabs for optimization, using PSI data.
* Finally, use Lighthouse CI on pull requests to ensure there are no regressions in Core Web Vitals before you deploy a change to production.

With that introduction, let's dive into the specific updates for each tool! 

### Lighthouse

Lighthouse is an automated website auditing tool that helps developers diagnose issues and identify opportunities to improve the user experience of their sites. It measures several dimensions of user experience quality in a lab environment, including performance and accessibility. The latest version of Lighthouse ([6.0](/lighthouse-whats-new-6.0/), released mid-May 2020) includes additional audits, new metrics, and a newly composed performance score.

<figure class="w-figure">
  <img class="w-screenshot" src="./lighthouse.png" alt="Lighthouse 6.0 showing the latest Core Web Vitals metrics">
</figure>

> Lighthouse 6.0 introduces three new metrics to the report. Two of these new metrics - [Largest Contentful Paint](/lcp/) (LCP) and [Cumulative Layout Shift](/cls/) (CLS) - are lab implementations of Core Web Vitals and provide important diagnostic information for optimizing user experience. Given their importance for assessing user experience, the new metrics are not only measured and included in the report, they are also factored in calculating the performance score.
> 
> The third new metric included in Lighthouse - [Total Blocking Time](/tbt/) (TBT) - correlates well with the field metric [First Input Delay](/fid/) (FID), another Core Web Vitals metric. Following the recommendations provided in the Lighthouse report and optimizing against your scores sets you up to provide the best possible experience to your users. 

Reviewer 2: ```suggestion
Lighthouse 6.0 introduces three new metrics to the report. Two of these new metrics—[Largest Contentful Paint](/lcp/) (LCP) and [Cumulative Layout Shift](/cls/) (CLS)—are lab implementations of Core Web Vitals and provide important diagnostic information for optimizing user experience. Given their importance for assessing user experience, the new metrics are not only measured and included in the report, they are also factored in calculating the performance score.

The third new metric included in Lighthouse—[Total Blocking Time](/tbt/) (TBT)—correlates well with the field metric [First Input Delay](/fid/) (FID), another Core Web Vitals metric. Following the recommendations provided in the Lighthouse report and optimizing against your scores sets you up to provide the best possible experience to your users. 
```
